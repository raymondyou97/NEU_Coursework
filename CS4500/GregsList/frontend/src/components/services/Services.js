import React, { Component } from 'react';

import '../../styles/services/services.css';

class Services extends Component {
  componentDidMount() {
    this.props.findAllServices();
  }

  deleteService = id => {
    this.props.deleteService(id);
  };

  enableEdits = (id, title) => {
    this.props.enableEdits(id, title);
  };

  handleTitleChange = e => {
    this.props.handleTitleChange(e.target.value);
  };

  showService = id => {
    this.props.showService(id);
  };

  render() {
    const { newTitle } = this.props;
    return (
      <div>
        <h3>Services</h3>
        <table className="service-table">
          <tbody className="service-table-body">
            <tr className="service-columns">
              <td>Title</td>
            </tr>
            <tr className="service-inputs">
              <td>
                <input
                  className="service-input-title"
                  type="text"
                  placeholder="Title"
                  value={newTitle}
                  onChange={this.handleTitleChange}
                />
              </td>
              <td className="service-input-buttons">
                <button
                  className="service-add-button"
                  onClick={() => this.props.createService()}
                >
                  Add
                </button>
                <button
                  className="service-update-button"
                  onClick={() => this.props.updateService()}
                >
                  Update
                </button>
              </td>
            </tr>
            {this.props.services.map(service => (
              <tr className="service-details" key={service.id}>
                <td onClick={() => this.showService(service.id)}>
                  {service.title}
                </td>
                <td className="service-buttons">
                  <button
                    className="service-delete-button"
                    onClick={() => this.deleteService(service.id)}
                  >
                    Delete
                  </button>
                  <button
                    className="service-edit-button"
                    onClick={() => this.enableEdits(service.id, service.title)}
                  >
                    Edit
                  </button>
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

export default Services;
