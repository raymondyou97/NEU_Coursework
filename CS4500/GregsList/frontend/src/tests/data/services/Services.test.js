'use es6';

import React from 'react';
import Services from '../../../components/services/Services';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { SERVICE_LIST } from './ServiceData';

configure({ adapter: new Adapter() });

test('Snapshot renders', () => {
  const component = renderer.create(
    <Services
      newTitle={''}
      findAllServices={() => {}}
      enableEdits={() => {}}
      services={SERVICE_LIST}
      handleTitleChange={() => {}}
      createService={() => {}}
      showService={() => {}}
      deleteService={() => {}}
      updateService={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Services component updates when there is service data', () => {
  const component = renderer.create(
    <Services
      newTitle={''}
      findAllServices={() => {}}
      enableEdits={() => {}}
      services={SERVICE_LIST}
      handleTitleChange={() => {}}
      createService={() => {}}
      showService={() => {}}
      deleteService={() => {}}
      updateService={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Initial DOM contains correct number of rows', () => {
  const component = shallow(
    <Services
      newTitle={''}
      findAllServices={() => {}}
      enableEdits={() => {}}
      services={SERVICE_LIST}
      handleTitleChange={() => {}}
      createService={() => {}}
      showService={() => {}}
      deleteService={() => {}}
      updateService={() => {}}
    />
  );

  expect(component.find('tr').length).toEqual(5);
});

test('Services component input value changes if selected title', () => {
  const component = shallow(
    <Services
      newTitle={'Dog Walking'}
      findAllServices={() => {}}
      enableEdits={() => {}}
      services={SERVICE_LIST}
      handleTitleChange={() => {}}
      createService={() => {}}
      showService={() => {}}
      deleteService={() => {}}
      updateService={() => {}}
    />
  );

  expect(component.find('.service-input-title').props().value).toEqual(
    'Dog Walking'
  );
});
