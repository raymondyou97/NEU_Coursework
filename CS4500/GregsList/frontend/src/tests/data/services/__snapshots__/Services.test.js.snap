// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services component updates when there is service data 1`] = `
<div>
  <h3>
    Services
  </h3>
  <table
    className="service-table"
  >
    <tbody
      className="service-table-body"
    >
      <tr
        className="service-columns"
      >
        <td>
          Title
        </td>
      </tr>
      <tr
        className="service-inputs"
      >
        <td>
          <input
            className="service-input-title"
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value=""
          />
        </td>
        <td
          className="service-input-buttons"
        >
          <button
            className="service-add-button"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="service-update-button"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-details"
      >
        <td
          onClick={[Function]}
        >
          Catering
        </td>
        <td
          className="service-buttons"
        >
          <button
            className="service-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-details"
      >
        <td
          onClick={[Function]}
        >
          Dog Walking
        </td>
        <td
          className="service-buttons"
        >
          <button
            className="service-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-details"
      >
        <td
          onClick={[Function]}
        >
          Petsitting
        </td>
        <td
          className="service-buttons"
        >
          <button
            className="service-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Snapshot renders 1`] = `
<div>
  <h3>
    Services
  </h3>
  <table
    className="service-table"
  >
    <tbody
      className="service-table-body"
    >
      <tr
        className="service-columns"
      >
        <td>
          Title
        </td>
      </tr>
      <tr
        className="service-inputs"
      >
        <td>
          <input
            className="service-input-title"
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value=""
          />
        </td>
        <td
          className="service-input-buttons"
        >
          <button
            className="service-add-button"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="service-update-button"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-details"
      >
        <td
          onClick={[Function]}
        >
          Catering
        </td>
        <td
          className="service-buttons"
        >
          <button
            className="service-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-details"
      >
        <td
          onClick={[Function]}
        >
          Dog Walking
        </td>
        <td
          className="service-buttons"
        >
          <button
            className="service-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-details"
      >
        <td
          onClick={[Function]}
        >
          Petsitting
        </td>
        <td
          className="service-buttons"
        >
          <button
            className="service-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
